test/Triton/ops.mlir

```diff
@@ -250,3 +250,10 @@ tt.func @experimental_descriptor_load(%0: !tt.tensordesc<tensor<128xf32>>) {
   %1 = tt.experimental_descriptor_load %0[%c0_i32] : !tt.tensordesc<tensor<128xf32>> -> tensor<128xf32>
   tt.return
 }
+
+// CHECK-LABEL: @gather_op
+tt.func @gather_op(%arg0: tensor<128x16xf32>, %arg1: tensor<512x4xi32>) -> tensor<512x4xf32> {
+  // CHECK-NEXT: %0 = tt.gather %arg0[%arg1] {axis = 0 : i32} : (tensor<128x16xf32>, tensor<512x4xi32>) -> tensor<512x4xf32>
+  %0 = tt.gather %arg0[%arg1] {axis = 0 : i32} : (tensor<128x16xf32>, tensor<512x4xi32>) -> tensor<512x4xf32>
```

Comment:
just starting to look at this but shouldn't the index tensor be a 1D tensor if we index only along 1 dimension?

Reply:
Gather along a single axis `d` means that each column along dim `d` in the output is comprised of the elements from the corresponding column in the source tensor. E.g. `out[i,j] = src[idx[i,j],j]` for `axis=0`.

Reply:
ah I see, ok yeah looks like that's what pytorch. I'll let @apgoucher confirm that it is what he wants but makes sense to me.